File: iosApp/iosApp/feature/profile/order/order_details/OrderDetailsViewState.swift
//
//  OrderDetailsViewState.swift
//  iosApp
//
//  Created by Марк Шавловский on 02.02.2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import shared
import SwiftUI


struct OrderDetailsViewState {
    let orderUuid:String
    let orderProductItemList:[OrderProductItem]
    let deliveryCost: String?
    let newTotalCost:String
    let state: OrderDetailsDataState.ScreenState
    let code: String
    let orderInfo: OrderInfo?
    let discount:String?
}

struct OrderInfo{
    let status: OrderStatus
    let statusName: String
    let dateTime: String
    let deferredTime: String?
    let address: String
    let comment: String?
    let pickupMethod: String
    let deferredTimeHint: String
    let paymentMethod: LocalizedStringKey?
}

File: feature/order/src/commonMain/kotlin/com/bunbeauty/order/ui/screen/orderdetails/OrderDetailsViewState.kt
package com.bunbeauty.order.ui.screen.orderdetails

import com.bunbeauty.core.model.order.OrderStatus
import com.bunbeauty.core.base.BaseViewState
import com.bunbeauty.order.presentation.order_details.OrderDetails

data class OrderDetailsViewState(
    val orderUuid: String,
    val orderProductItemList: List<OrderProductUiItem>,
    val deliveryCost: String?,
    val newTotalCost: String,
    val state: OrderDetails.DataState.ScreenState,
    val code: String,
    val orderInfo: OrderInfo?,
    val discount: String?,
) : BaseViewState {
    data class OrderInfo(
        val status: OrderStatus,
        val statusName: String,
        val dateTime: String,
        val deferredTime: String?,
        val address: String,
        val comment: String?,
        val pickupMethod: String,
        val deferredTimeHint: String,
        val paymentMethod: String?,
    )
}
